### PR TITLE
[macOS] Add gcc 12

### PR DESCRIPTION
**Pull request by sergei-pyshnoi**
# Description
Add gcc 12 to macos-12, macos-13 images

#### Related issue:

https://github.com/actions/runner-images/issues/6969

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/toolsets/toolset-11.json [hunk 1]
```diff
@@ -233,9 +233,9 @@
     },
     "gcc": {
         "versions": [
-            "9",
```
[the remark this answers is not in the document]
Comment (sergei-pyshnoi) [reply images/macos/toolsets/toolset-11.json:236]: @mikhailkoliada, return the 9 ver back

**Comment by mathomp4** [comment]
Huh. So this is how it is done. Well, come GCC 13 time, I can try the PR myself! 😄

**Review by mikhailkoliada — APPROVED** [review]

**Pull request merged**